### DISCUSSION TITLE
Retention period for CW Logs in cloudtrial/periodic mode?

### DISCUSSION THREAD

**Discussion by danielhros** [cloud-custodian/cloud-custodian#9685] [AWS] [answered]
Is it possible to set the retention period for CW Logs? I haven't found such a thing in the documentation or discussions, and when I deploy the CC policy in periodic or cloudtrail mode, the retention is set to never expire.

**Comment by ajkerrigan** [comment, marked as answer]
Not at deploy time, since Custodian doesn't create those log groups when you provision a policy. They're created under the hood when a function runs. Since policy Lambda function log groups have a predictable naming pattern (generally `/aws/lambda/custodian-<policy name>`) though, one approach is to run a policy with the [retention action](https://cloudcustodian.io/docs/aws/resources/log-group.html#aws-log-group-actions-retention) to configure retention for any log groups with Custodian's naming pattern.

**Reply by danielhros** [reply]
Thanks for reply :)
